viewing_party/party.py

```diff
@@ -1,23 +1,136 @@
 # ------------- WAVE 1 --------------------
 
 def create_movie(title, genre, rating):
-    pass
+    if title and genre and rating:
+        return {
+            "title": title,
+            "genre": genre,
+            "rating": rating
+            }
+    
+    return None
 
+def add_to_watched(user_data, movie):
+    if movie not in user_data.get("watched"):
+        user_data.get("watched").append(movie)
+
+    return user_data
+
+def add_to_watchlist(user_data, movie):
+    if movie not in user_data.get("watchlist"):
+        user_data.get("watchlist").append(movie)
+
+    return user_data
+
+def watch_movie(user_data, title):
+    movies = user_data.get("watchlist")
+
+    for movie in movies:
+        if title == movie["title"]:
+            movies.remove(movie)
+            user_data.get("watched").append(movie)
+    
+    return user_data
 # -----------------------------------------
 # ------------- WAVE 2 --------------------
 # -----------------------------------------
+def get_watched_avg_rating(user_data):
+    rating = []
+    try:
+        for movie in user_data["watched"]:
+            rating.append(movie["rating"])
+        return sum(rating) / len(rating)
+    except ZeroDivisionError:
+        return 0.0
+    except KeyError: # Incase the key watched is missing
+        return None
 
+def get_most_watched_genre(user_data):
+    genre = {}
 
+    try:
+        for movie in user_data["watched"]:
+            if movie["genre"] not in genre:
+                genre[movie["genre"]] = 1
+
+            genre[movie["genre"]] += 1
+            print(genre)
+    except KeyError: # Incase the key watched is missing
+        return None
+    
+    if genre:
+        return max(genre, key=genre.get)
+    
+    return None
 # -----------------------------------------
 # ------------- WAVE 3 --------------------
-# -----------------------------------------
+# ------------------------------------------
+from frozendict import frozendict

```

Comment:
Generally speaking we want to have import statements at the top of the file. There are a few exceptions to this, but it's a general rule.